source/dataman/CacheMan.cpp

```diff
@@ -44,25 +44,20 @@ int CacheItem::put(const void *a_data, json a_jmsg)
     size_t varbytes = a_jmsg["varbytes"].get<size_t>();
     size_t dsize = a_jmsg["dsize"].get<size_t>();
 
-    size_t timestep = 0;
-
     if (m_cache.empty())
     {
         push();
     }
 
-    if (a_jmsg["timestep"].is_number())
-    {
-        timestep = a_jmsg["timestep"].get<size_t>();
-    }
-
     for (size_t i = 0; i < putsize; i += chunksize)
     {
         std::vector<size_t> p = one2multi(putshape, i);
         p = apply_offset(p, offset);
         size_t ig = multi2one(varshape, p);
-        std::copy((char *)a_data + i * dsize,
-                  (char *)a_data + i * dsize + chunksize * dsize,
+        std::copy(const_cast<char *>(static_cast<const char *>(a_data)) +
```

Comment:
The input iterators here should not be un-const-'d.  The input components to almost any STL algorithm should be const.  The general idea is to avoid using const_cast unless you absolutely have to as it's usage usually means there's something borked in the API usage (not always of course, but usually).  An example might be if you're using a library what doesn't have any const correctness, like ZFP, where even the input data is non-const.

In this case, however, the input pointers can stay const.